src/test/java/bowlingexercise3/BowlingTest.java

```diff
@@ -61,22 +81,339 @@ public void testBowlingSpares(String turns, int score) {
+            "1+|11|11|11|11|11|11|11|11|11||",
+            "111|11|11|11|11|11|11|11|11|11||",
+            "11|11|11|11|11|11|11|11|11|111||",
+            "11|11||11|11|11|11|11|11|11|11|||",
+            "11|11|11|11|11|1X|11|11|11|11||",
+            "X1|11|11|11|11|11|11|11|11|11||",
+            "11|11|10|11|11|11|11|11|11|11||",
+            "//|11|11|11|11|//|11|11|11|11||",
+            "11|11|11|11|11|11|11|11|-X|11||",
+            "///|11|22|11|11|//|11|11|11|11||",
+            "11|11|45|11|11|11|11|11|-X|11||",
+            "X|X|X|X|X|X|X|X|X|XX||",
+            "X|X|X|X|X|X|X|X|X|XXX||",
+            "5|X|X|X|X|X|X|X|X|X||",
+            "X|X|X|X|X|X|X|X|X|/||",
+            "X|X|X|X|X|X|X|X|X|6||XX",
+            "X|X|X|X|X|X|X|X|X|?||35",
+            "X|X|X|X|X|X|X|X|X|X||81-",
+            "9|X|X|X|X|X|2X|X|X|X||",
+            "9|X|X|X|X|X|X|X|X|X||",
+            "/X|X|X|X|X|X|X|X|X|X||",
+            "-/|X|X|X|X|X|X|X|X|X||-X",
+
+
+    })
+    public void TestingInvalidInput(String turns) {
+
+        Bowling game = new Bowling();
+        Exception e = assertThrows(IllegalArgumentException.class, () -> game.turns(turns));
+        assertEquals("Input isn't valid", e.getMessage());
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {
+
+            "11|11|11|11|11|11|11|11|11|11||",
+            "23|32|11|11|11|11|11|11|11|11||",
+            "45|54|11|11|11|11|11|11|11|11||",
+            "63|36|11|11|11|11|11|11|11|11||",
+            "72|27|11|11|11|11|11|11|11|11||",
+            "81|18|11|11|11|11|11|11|11|11||",
+            "81|18|11|11|11|11|11|11|11|11||",
+            "81|1-|-1|11|11|11|11|-1|11|11||",
+            "--|--|--|--|--|--|--|--|--|--||",
+            "22|22|22|22|22|22|22|22|22|22||",
+            "33|33|33|33|33|33|33|33|33|33||",
+            "44|44|44|44|44|44|44|44|44|44||",
+            "12|12|12|12|12|12|12|12|12|12||",
+            "21|21|21|21|21|21|21|21|21|21||",
+            "18|18|18|18|18|18|18|18|18|18||",
+            "72|72|72|72|72|72|72|72|72|72||",
+            "63|63|63|63|63|63|63|63|63|63||",
+            "54|54|54|54|54|54|54|54|54|54||",
+            "22|22|45|63|72|81|11|11|11|11||",
+            "11|54|11|11|11|11|11|11|11|11||",
+            "18|54|61|71|34|35|6-|-8|-4|17||",
+            "54|54|55|54|55|54|54|54|54|54||",
+            "45|45|45|45|45|45|45|45|45|45||",
+            "18|9-|9-|9-|9-|-9|-9|9-|9-|9-||",
+            "1/|1/|1/|1/|1/|1/|1/|1/|1/|1/||",
+            "-/|-/|-/|-/|-/|-/|-/|-/|-/|-/||",
+            "2/|2/|2/|2/|2/|2/|2/|2/|2/|2/||",
+            "3/|3/|3/|3/|3/|3/|3/|3/|3/|3/||",
+            "4/|4/|4/|4/|4/|4/|4/|4/|4/|4/||",
+            "5/|5/|5/|5/|5/|5/|5/|5/|5/|5/||",
+            "6/|6/|6/|6/|6/|6/|6/|6/|6/|6/||",
+            "7/|7/|7/|7/|7/|7/|7/|7/|7/|7/||",
+            "8/|8/|8/|8/|8/|8/|8/|8/|8/|8/||",
+            "9/|9/|9/|9/|9/|9/|9/|9/|9/|9/||",
+
+
+
+    })
+    public void testingValidInputNormalFrames(String turns) {
```

Comment:
normal - is that frames without strikes or spares?

Reply:
Yes, Normal implies no bonus ball